### PR TITLE
feat(TBD-10000): R2020-07: Support CDP Milestone II Dynamic Distro for CDP Data Center 7.x

### PR DESCRIPTION
**Pull request by Redwene**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.hadoop.distribution.cdp/src/main/java/org/talend/hadoop/distribution/cdp/CDP7xDistributionTemplate.java [hunk 1]
```diff
@@ -33,227 +34,253 @@
...
+	public boolean doSupportParquetOutput() {
+		return true;
+	}
+
+	@Override
+	public Set<ESparkVersion> getSparkVersions() {
+		Set<ESparkVersion> version = new HashSet<>();
+		Set<ESparkVersion> sparkVersions = super.getSparkVersions();
+		if (sparkVersions == null || sparkVersions.isEmpty()) {
+			version.add(ESparkVersion.SPARK_2_2);
```
Comment (AlixMetivier) [inline main/plugins/org.talend.hadoop.distribution.cdp/src/main/java/org/talend/hadoop/distribution/cdp/CDP7xDistributionTemplate.java:200]: should not it be spark 24 ?

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+			boolean storeByHBase = "true".equals(ElementParameterParser.getValue(node, "__STORE_BY_HBASE__"));
+		    
+
+			if(impalaDriver!=null && !"".equals(impalaDriver.trim()) && (this.isCustom || this.hiveDistrib.doSupportHive2())) {
+			    impalaDriver = impalaDriver.toLowerCase();
+				if ("hive2".equalsIgnoreCase(impalaDriver)) {
+					javaDbDriver = "org.apache.hive.jdbc.HiveDriver";
+					impalaDbProtocol="hive2";
+				}
+				if ("impala40".equalsIgnoreCase(impalaDriver)) {
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet:60]: can not be the case, you removed IMPALA40 and IMPALA41 to be IMPALA

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+					javaDbDriver = "com.cloudera.impala.jdbc4.Driver";
+					impalaDbProtocol="impala";
+				}	
+				if ("impala41".equalsIgnoreCase(impalaDriver)) {
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet:64]: can not be the case, you removed IMPALA40 and IMPALA41 to be IMPALA

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet [hunk 4]
```diff
@@ -72,7 +72,7 @@
 			    impalaDriver = impalaDriver.toLowerCase();
 				if ("hive2".equalsIgnoreCase(impalaDriver)) {
 					javaDbDriver = "org.apache.hive.jdbc.HiveDriver";
-					impalaDbProtocol="hive2";
+					impalaDbProtocol="impala";
 				}
 				if ("impala40".equalsIgnoreCase(impalaDriver)) {
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet:77]: can not be the case, you removed IMPALA40 and IMPALA41 to be IMPALA

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml [hunk 5]
```diff
@@ -74,7 +72,7 @@
 		
 		<PARAMETER NAME="USER" FIELD="TEXT" NUM_ROW="30"
 			REPOSITORY_VALUE="USERNAME" GROUP="CONNECTION" SHOW="true">
-			<DEFAULT>""</DEFAULT>
+			<DEFAULT></DEFAULT>
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml:75]: why this change, is it for URI creation ?

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtilImpalaDriver.javajet [hunk 6]
```diff
@@ -0,0 +1,298 @@
+<%
+
+	class ConnectionUtilImpala extends DefaultConnectionUtil {
+      // use HiveComponent type to get all necessary methods but there is something to fix about the design on the current typing 
+		private org.talend.hadoop.distribution.component.HiveComponent distrib = null;
+		private boolean isCustom;
+    
+		public ConnectionUtilImpala(org.talend.hadoop.distribution.component.HadoopComponent distrib) {
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtilImpalaDriver.javajet:8]: org.talend.hadoop.distribution.component.HadoopComponent can be changed to org.talend.hadoop.distribution.component.HiveComponent to avoid the cast

File: main/plugins/org.talend.hadoop.distribution.cdp/resources/builtin/cdp/Cloudera_CDP_7_1_1.json [hunk 7]
```diff
@@ -4702,90 +4534,18 @@
...
-        "childNodes" : [ ],
-        "id" : "DYNAMIC_Cloudera_CDP7_1_1_0_565_snakeyaml_1_24_jar",
-        "tagName" : "library"
       }, {
         "childNodes" : [ ],
         "id" : "DYNAMIC_Cloudera_CDP7_1_1_0_565_snappy_java_1_1_7_5_jar",
         "tagName" : "library"
-      }, {
-        "childNodes" : [ ],
-        "id" : "DYNAMIC_Cloudera_CDP7_1_1_0_565_spark_core_2_11_2_4_0_7_1_1_0_565_jar",
```
Comment (lbourgeois) [inline main/plugins/org.talend.hadoop.distribution.cdp/resources/builtin/cdp/Cloudera_CDP_7_1_1.json:4763]: I wonder why those Spark libs are not needed anymore and how they can't miss at job compilation

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet (hunk 2 above)
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet:60]: yes old code copy/paste. I will fix it

File: main/plugins/org.talend.hadoop.distribution.cdp/src/main/java/org/talend/hadoop/distribution/cdp/CDP7xDistributionTemplate.java (hunk 1 above)
Comment (Redwene) [reply main/plugins/org.talend.hadoop.distribution.cdp/src/main/java/org/talend/hadoop/distribution/cdp/CDP7xDistributionTemplate.java:200]: yes will fix it

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet (hunk 3 above)
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet:64]: yes old code copy/paste. I will fix it

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet (hunk 4 above)
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet:77]: yes old code copy/paste. I will fix it

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml (hunk 5 above)
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml:75]: I will fix it

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaCreateTable/tImpalaCreateTable_java.xml [hunk 8]
```diff
@@ -65,9 +65,7 @@
 			<ITEMS DEFAULT="HIVE2">
 				<ITEM NAME="HIVE2" VALUE="HIVE2"
 					  SHOW_IF="(HIVE_SERVER == 'HIVE2')" />
-				<ITEM NAME="IMPALA40" VALUE="IMPALA40"
-					  SHOW_IF="(DISTRIBUTION=='CLOUDERA' AND DISTRIB[DISTRIBUTION, IMPALA_VERSION].doSupportImpalaConnector[])" />
-				<ITEM NAME="IMPALA41" VALUE="IMPALA41"
+				<ITEM NAME="IMPALA" VALUE="IMPALA"
```
[the remark this answers is not in the document]
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaCreateTable/tImpalaCreateTable_java.xml:68]: in distribution before CDP this item it's never used

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtilImpalaDriver.javajet (hunk 6 above)
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtilImpalaDriver.javajet:8]: yes it could, but the constructor for impala using impala driver looks strange with HiveComponent. I did this choice to mask hiveComponent  (private) and for easier future changes

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet [hunk 9]
```diff
@@ -72,7 +72,7 @@
 			    impalaDriver = impalaDriver.toLowerCase();
 				if ("hive2".equalsIgnoreCase(impalaDriver)) {
 					javaDbDriver = "org.apache.hive.jdbc.HiveDriver";
-					impalaDbProtocol="hive2";
+					impalaDbProtocol="impala";
```
[the remark this answers is not in the document]
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet:75]: it's what we discussed,old code with dirty changes to work with impala driver. that why I'm still on it  to change this

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/component/ImpalaComponent.java [hunk 10]
```diff
@@ -18,17 +18,6 @@
  */
 public interface ImpalaComponent extends HadoopComponent {
 
-    /**
-     * indicate if we support the impala native protocol.
-     * connection with jdbc:impala://[Host]:[Port]/[Schema];[Property1]=[Value];[Property2]=[Value];
-     * @return
-     */
-    default boolean doSupportImpalaConnector() {
```
[the remark this answers is not in the document]
Comment (Redwene) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/component/ImpalaComponent.java:26]: I moved it to Haddo Component to find out if its impala is supported before converting the distribution object to impala. We can thus refactor more easily the code between the choice of the driver hive and impala.

File: main/plugins/org.talend.hadoop.distribution.cdp/src/main/java/org/talend/hadoop/distribution/cdp/CDP7xDistributionTemplate.java [hunk 11]
```diff
@@ -33,227 +34,253 @@
...
-    }
-    @Override
-    public KuduVersion getKuduVersion() {
-        return KuduVersion.KUDU_1_12;
-    }
+public class CDP7xDistributionTemplate extends AbstractDynamicCDPDistributionTemplate
+		implements HDFSComponent, HBaseComponent, HCatalogComponent, MRComponent, HiveComponent, HiveOnSparkComponent,
+		ImpalaComponent, SqoopComponent, CDPSparkBatchComponent, SparkStreamingComponent, ICDP7xDistributionTemplate {
+	private final static String SEPARATOR = ",";
+	public final static String DEFAULT_LIB_ROOT = "/opt/cloudera/parcels/CDH/lib";
```
[the remark this answers is not in the document]
Comment (Redwene) [reply main/plugins/org.talend.hadoop.distribution.cdp/src/main/java/org/talend/hadoop/distribution/cdp/CDP7xDistributionTemplate.java:41]: yes this is cloudera standard path

File: main/plugins/org.talend.hadoop.distribution.cdp/resources/builtin/cdp/Cloudera_CDP_7_1_1.json (hunk 7 above)
Comment (Redwene) [reply main/plugins/org.talend.hadoop.distribution.cdp/resources/builtin/cdp/Cloudera_CDP_7_1_1.json:4763]: we are in the IMPALA-LIB-DYNAMIC section. this pot is loaded elsewhere.

File: main/plugins/org.talend.hadoop.distribution.cdp/src/main/java/org/talend/hadoop/distribution/cdp/CDP7xDistributionTemplate.java (hunk 1 above)
Comment (Redwene) [reply main/plugins/org.talend.hadoop.distribution.cdp/src/main/java/org/talend/hadoop/distribution/cdp/CDP7xDistributionTemplate.java]: done

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet (hunk 2 above)
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet]: done

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet (hunk 9 above)
[the remark this answers is not in the document]
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet]: done I quick fix for ssl, I will do the same for keberos.

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet (hunk 3 above)
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnectionUtil.javajet]: done

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet (hunk 4 above)
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet]: done

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml (hunk 5 above)
Comment (Redwene) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml]: done

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaCreateTable/tImpalaCreateTable_java.xml (hunk 8 above)
[the remark this answers is not in the document]
Comment (lbourgeois) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaCreateTable/tImpalaCreateTable_java.xml:68]: My understanding is that tImpalaCreateTable component is also available for other distros bit if that's not the case lgtm
